### PR TITLE
TST: silence test failures on macOS for `beta.ppf` overflow

### PR DESCRIPTION
**Pull request by rgommers**
See gh-14901 for details.

[skip azp]
[skip github]

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -37,6 +37,10 @@
 # python -OO strips docstrings
 DOCSTRINGS_STRIPPED = sys.flags.optimize > 1
 
+# Failing on macOS 11, Intel CPUs. See gh-14901
+MACOS_INTEL = (sys.platform == 'darwin') and (platform.machine() == 'x86_64')
```
Comment (tylerjereddy) [inline scipy/stats/tests/test_distributions.py:41]: I think we need to add `import platform` above--let me see if I can push that in real quick

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -2879,6 +2886,7 @@ def test_issue_12796(self):
         res = stats.beta.cdf(inv, a, b)
         assert_allclose(res, 1 - alpha_2)
 
+    @pytest.mark.skipif(MACOS_INTEL, reason="Overflow, see gh-14901")
```
Comment (tylerjereddy) [inline scipy/stats/tests/test_distributions.py:2889]: I thought the third failing test was `test_issue_12635` above, unless I'm misreading the diff

```
 FAILED scipy/stats/tests/test_distributions.py::TestBeta::test_issue_12635 - ...
FAILED scipy/stats/tests/test_distributions.py::TestBeta::test_issue_12794 - ...
FAILED scipy/stats/tests/test_distributions.py::TestBeta::test_issue_12796 - ...
```

Comment (tylerjereddy) [reply scipy/stats/tests/test_distributions.py]: I went ahead and pushed that in--probably just some speedy copy-paste work.

**Comment by tylerjereddy** [comment]
I'll go ahead and squash merge since the MacOS CI entries are now passing and my correction commits hardly need to be separated for this temporary bypass PR.

**Comment by rgommers** [comment]
Thanks Tyler!

**Pull request merged**